### PR TITLE
Extention template: Enable DuckDB-Wasm extensions

### PR DESCRIPTION
**Pull request by carlopi**

### CODE REVIEW

**Comment by Mytherin** [comment]
@samansmink could you have a look?

**Comment by samansmink** [comment]
To review this I would like to have a draft PR from the extension template that shows that this workflow works. Have already discussed this with @carlopi.

**Comment by carlopi** [comment]
Eventual successful run will be here: https://github.com/carlopi/extension-template/actions/runs/6959995349.

This includes an hack to workaround both a limitation of the parsing script AND the fact that in the source code for v0.9.2 wasm target is currently missing.

Hack can be potentially removed at some point, but should not generate any problem, since it's simply checking for file equality ( that meant something is already gone wrong).

**Comment by carlopi** [comment]
Successful run (on custom repo) is here: https://github.com/carlopi/extension-template/actions/runs/6959995349

File: .github/config/distribution_matrix.json [hunk 1]
```diff
@@ -39,5 +39,21 @@
         "vcpkg_triplet": "x64-windows-static-md"
       }
     ]
+  },
+  "wasm": {
```
Comment (samansmink) [inline .github/config/distribution_matrix.json:43]: Is there a reason there are 3 architectures here, but only 2 in the backport in [_extension_distribution.yml](https://github.com/duckdb/duckdb/pull/9356/files#diff-feabb7d2bbf4ed78e4061049b5a95bf92059972e8e132e391902b6cc148bff13) ?

**Review by samansmink — COMMENTED** [review]
Looks good, very excited to push for more wasm extensions!

File: .github/config/distribution_matrix.json (hunk 1 above)
Comment (carlopi) [reply .github/config/distribution_matrix.json:43]: Nicely spotted, I went back and forth on this, but probably it's simpler and better to have them uniform.

**Comment by carlopi** [comment]
Examples of uses of this:
* all target enabled: https://github.com/carlopi/extension-template/actions/runs/6971762883
* exclude a wasm arch (threads): https://github.com/carlopi/extension-template/actions/runs/6970358397/job/18968112690
* skip wasm entirely: https://github.com/carlopi/extension-template/actions/runs/6970355299

**Review by samansmink — APPROVED** [review]
looks good!

**Comment by carlopi** [comment]
Did some more testing, found a minor corner case that lead to even more proper handling of the special case, I think this is ready to be merged.

Tests are not really testing anything here.

**Comment by carlopi** [comment]
Even more bikeshedding.

I think this is properly ready and tested.

**Pull request merged**